src/types.ts

```diff
@@ -171,12 +180,13 @@ export interface IUnparsedCommunity extends IUnparsedMHBOListing {
 }
 
 export interface IRestResource<T> {
-  search: (params: ISearchParams) => Promise<T[]>
+  details: (params: ISearchParams) => Promise<T[]>
```

Comment:
This is a little backwards from what I was thinking. Maybe I confused things by defining searchSummary and emphasizing the detail_level, when that should be somewhat hidden from the api user.  I'd like the methods to reflect behavior instead of data. The actions are: searching, return entities by IDs, (and in the future), update, create, delete. Details and summaries are just part of the mechanics of how we are returning the different interface types.

It would be nice to be able have a good understanding of how the method works and what it does from the parameters and the return type themselves.

homes.searchSummary(ISearchParams): [IMHBOListing]
homes.byIds([int]): [IMobileHome]
   
We could have just changed homes.search to return IMHBOListing, but I thought we could keep the original search method for backward compatibility and by saying "searchSummary" it would be extra clear,  but I guess that had the opposite effect. The behavior I wanted was that search let's you know what matched(id + contextual info) and that byId give you the actual entity.